### ISSUE TITLE
Full-screen site switcher sets incorrect initial selection when loading a large number of sites

### ISSUE THREAD

**Issue by dcalhoun** [Automattic/wp-calypso#97342]
### Quick summary

When the _full-screen variant_ of the site switcher initially loads, it sets the initial selection to the first site listed. However, if the current user has many sites, when additional sites (lazily?) load, the site switcher erroneously moves selection to a site hidden outside of the currently visible sites in the scroll view. 

This results in unexpected or incorrect site selection when attempting to navigate with a keyboard. E.g., pressing the <kbd>Return</kbd> key selects the out-of-view site, and pressing the down arrow key dow not move selection to the first visible site in the list. 

The issue does not occur for accounts with a small number of sites, presumably because all of the sites are loaded in one step, rather than lazily loaded. It also only occurs for the full-screen version of the site switcher that is used when selecting _Switch site_ from a WP Admin page, the issue does not occur when using the _Switch site_ command from a WordPress.com page. 

| Affected | Unaffected |
| - | - |
| <img width="1282" alt="Image" src="https://github.com/user-attachments/assets/7e5f2db7-e786-431c-a43d-8477b660da04" /> | <img width="1282" alt="Image" src="https://github.com/user-attachments/assets/8566abd5-ebd4-4a9b-bb28-39f9c5d2615e" /> |

### Steps to reproduce

1. With an account the possesses many sites, visit a WordPress.com site's WP Admin _Dashboard_—e.g., `https://<SITE_DOMAIN>/wp-admin/`. 
2. Press <kbd>Command</kbd> + <kbd>K</kbd>. 
3. Select the site switcher command. 
4. Wait for the site list to load. 
5. Press the keyboard down arrow key. 

### What you expected to happen

The site switcher selection is moved to the second item. 

### What actually happened

The site switcher selection is moved a site somewhere much further down the list. 

### Impact

Some (< 50%)

### Available workarounds?

Yes, easy to implement

### If the above answer is "Yes...", outline the workaround.

Typing a search query causes the site switcher to refresh the current selection, resetting the current selection to the expected location. 

### Platform (Simple and/or Atomic)

Simple, Atomic, Self-hosted

### Logs or notes

macOS 15.1.1
Safari 18.1.1

**Comment by Robertght** [comment]
@dcalhoun I'm unable to replicate this on my end. Here's a short video:

https://github.com/user-attachments/assets/5edce64e-ea03-4258-8bb3-9520b12fce64

Could you share a recording of what you're seeing please?

**Comment by dcalhoun** [comment]
@Robertght thanks for following up. 🙇‍♂ 

From testing further, it appears the issue I describe only occurs for the full-screen variant of this UI, not the overlay variant. The full-screen variant is utilized if you begin the flow from a WP Admin page—e.g., starting from the WP Admin _Dashboard_, not the WordPress.com _My Home_. 

I updated the issue description with further details for reproducing the issue.

**Comment by janmckell** [comment]
@dcalhoun Hey! I tested this on Safari and on Chrome, on Mac OS 15.3.1. and it didn't occur for me. Can you confirm if it's still happening for you. It may be closed, if not.

Safari 18.3
Chrome 130.0.6723.117

**Comment by lsl** [comment]
@Automattic/t-rex is likely a better fit here.

> A small aside: I note that when the site switcher initially loads, the selected site background highlight color is blue, before switching to a magenta color once it finishes loading. I presume the former color is the desired color.

I see this but only on some users, probably only affects old users with account level color schemes set.

> I tested this on Safari and on Chrome, on Mac OS 15.3.1. and it did occur for me.

I wasn't able to repro this on firefox but using epiphany I noticed this behavior to do with the mouse placement, is this what you're seeing?

https://github.com/user-attachments/assets/c0740181-ae52-4e7e-969c-3f763ec626d2

What's happening here: if you hit the arrow key down it will jump to wherever the mouse is targeting.

**Comment by jartes** [comment]
I'm not able to reproduce this with two different WordPress.com accounts I have. Tried with Chrome and Safari:

https://github.com/user-attachments/assets/ad94243e-36fa-481a-851a-7bafd8dbe7df

**Comment by dsas** [comment]
I can reproduce as @lsl says using firefox.

https://github.com/user-attachments/assets/ee5d792c-585b-40cc-9f9d-3bcf0441a290

The wpcom command palette is used by approximately zero people per day, so a low priority is appropriate.
